data-model/src/main/java/io/micronaut/data/model/naming/NamingStrategy.java

```diff
@@ -110,7 +113,9 @@ public interface NamingStrategy {
      * @return The mapped name
      */
     default @NonNull String mappedName(Association association) {
-        String providedName = association.getAnnotationMetadata().stringValue(MappedProperty.class).orElse(null);
+        AnnotationMetadata assocationAnnotationMetadata = association.getAnnotationMetadata();
+        boolean generated = assocationAnnotationMetadata.booleanValue(MappedProperty.class, "generated").orElse(false);
+        String providedName = generated ? null : assocationAnnotationMetadata.stringValue(MappedProperty.class).orElse(null);
```

Comment:
Notice how the method is annotated `@NonNull`

Reply:
> Notice how the method is annotated `@NonNull`

Yes, this still should return non null value. It is just forcing to recalculate if it was generated, because we might need different strategy vs default which was the idea how to use Raw for Cosmos (or some similar example).

Reply:
OK my bad